Firmware/first_lay_cal.cpp

```diff
@@ -96,65 +107,61 @@ bool lay1cal_load_filament(uint8_t filament)
 //! @param extrusion_width the width of the extrusion layer
 void lay1cal_intro_line(bool extraPurgeNeeded, float layer_height, float extrusion_width)
 {
-    static const char cmd_intro_mmu_3[] PROGMEM = "G1 X55 E29 F1073";
-    static const char cmd_intro_mmu_4[] PROGMEM = "G1 X5 E29 F1800";
-    static const char cmd_intro_mmu_5[] PROGMEM = "G1 X55 E8 F2000";
-    static const char cmd_intro_mmu_6[] PROGMEM = "G1 Z0.3 F1000";
-    static const char cmd_intro_mmu_8[] PROGMEM = "G1 X240 E25  F2200";
-    static const char cmd_intro_mmu_9[] PROGMEM = "G1 Y-2 F1000";
-    static const char cmd_intro_mmu_10[] PROGMEM = "G1 X55 E25 F1400";
-    static const char cmd_intro_mmu_11[] PROGMEM = "G1 Z0.2 F1000";
-    static const char cmd_intro_mmu_12[] PROGMEM = "G1 X5 E4 F1000";
-
-    static const char * const intro_mmu_cmd[] PROGMEM =
+    static const char cmd_intro_mmu_3[] PROGMEM = "G1X55E29F1073";
+    static const char cmd_intro_mmu_4[] PROGMEM = "G1X5E29F1800";
+    static const char cmd_intro_mmu_5[] PROGMEM = "G1X55E8F2000";
+    static const char cmd_intro_mmu_6[] PROGMEM = "G1Z0.3F1000";
+    static const char cmd_intro_mmu_8[] PROGMEM = "G1X240E25F2200";
+    static const char cmd_intro_mmu_9[] PROGMEM = "G1Y-2F1000";
+    static const char cmd_intro_mmu_10[] PROGMEM = "G1X200E8F1400";
+    static const char cmd_intro_mmu_11[] PROGMEM = "G1Z0.2F1000";
+    static const char * const cmd_intro_mmu[] PROGMEM =
     {
         // first 2 items are only relevant if filament was not loaded - i.e. extraPurgeNeeded == true
         cmd_intro_mmu_3,
         cmd_intro_mmu_4,
-
         cmd_intro_mmu_5,
         cmd_intro_mmu_6,
         zero_extrusion,
         cmd_intro_mmu_8,
         cmd_intro_mmu_9,
         cmd_intro_mmu_10,
         cmd_intro_mmu_11,
-        cmd_intro_mmu_12,
     };
 
     if (MMU2::mmu2.Enabled())
     {
-        for (uint8_t i = (extraPurgeNeeded ? 0 : 2); i < (sizeof(intro_mmu_cmd)/sizeof(intro_mmu_cmd[0])); ++i)
+        for (uint8_t i = (extraPurgeNeeded ? 0 : 2); i < (sizeof(cmd_intro_mmu)/sizeof(cmd_intro_mmu[0])); ++i)
         {
-            enquecommand_P(static_cast<char*>(pgm_read_ptr(&intro_mmu_cmd[i])));
+            enquecommand_P(static_cast<char*>(pgm_read_ptr(&cmd_intro_mmu[i])));
         }
     }
     else
     {
-        static const char fmt1[] PROGMEM = "G1 X%d E%-.3f F1000";
-        enquecommandf_P(fmt1, 60, count_e(layer_height, extrusion_width * 4.f, 60));
-        enquecommandf_P(fmt1, 100, count_e(layer_height, extrusion_width * 8.f, 40));
+        enquecommandf_P(extrude_fmt_X, (float)60, count_e(layer_height, extrusion_width * 4.f, 60));
+        enquecommandf_P(extrude_fmt_X, (float)202.5, count_e(layer_height, extrusion_width * 8.f, 142.5));
```

Comment:
This new format string removed the feedrate settings, previously it was 1000 mm/min or 16.67 mm/s

This previous g-code is setting 1000 mm/min so in this case it's OK. But if that string is changed, then this g-code will also change with it.

Reply:
If no feedrate is provided the last set feedrate is used. Saves few bytes and the printed lines look the same as before.

Reply:
I agree with @gudnimg that we should probably set the feedrate to something known here once before the non-mmu purge line. No clue if something before it sets the feedrate to 1000. And yea, not specifying the F every command is fine, but we should still set the F before these two lines just for consistency at the expense of ~30B of flash.
From what I see quickly looking through the code, nothing before this sets the feedrate, so these two lines would run at whatever feedrate was set before this, which could be reasonable (default feedrate) or something ridiculously high if something else executed before it. Please enqueue a `G1F<something>` before enqueueing any other commands in the non-mmu intro.

Reply:
Pushed, please double check.

Reply:
Another minor nitpick.

You can change:

`(float)60` to `60.f`
`(float)202.5` to` 202.5f`

Slightly shorter code.

Reply:
@gudnimg fixed nitpick